jetty-servlet/src/test/java/org/eclipse/jetty/servlet/ServletContextHandlerTest.java

```diff
@@ -1651,21 +1651,35 @@ public void testProgrammaticFilterServlet() throws Exception
         ServletHandler handler = new ServletHandler();
         _server.setHandler(context);
         context.setHandler(handler);
-        handler.addServletWithMapping(new ServletHolder(new TestServlet()), "/");
+        handler.addServletWithMapping(new ServletHolder(new TestServlet()
+        {
+            @Override
+            public void init() throws ServletException
+            {
+                handler.addFilterWithMapping(new FilterHolder(new MyFilter()), "/test/*", EnumSet.of(DispatcherType.REQUEST));
+            }
+        }), "/test");
```

Comment:
Can we also please add tests for all the types of objects: adding servlets, filters and listeners from servlets, filters and listeners to make sure that the jetty apis are doing what we expect